adminSiteClient/GrapherConfigGridEditorTypesAndUtils.tsx

```diff
@@ -339,6 +340,38 @@ export function SExpressionToJsonLogic(
     })
 }
 
+/** JsonLogic is the easiest format that the React Awesome Query Library can round
+    trip (i.e. deserialize from). Building the internal structure of the query library
+    would be tedious so we convert our SExpressions to JsonLogic. When React Awesome
+    Query Library parses this, we need to convert some custom things like the is_latest
+    query operator that we added to RAQL that in our SExpressions are represented as
+    field = "latest"
+ */
+export function postProcessJsonLogicTree(filterTree: JsonTree | JsonItem) {
+    if (filterTree.type === "group" && filterTree.children1) {
+        if (isArray(filterTree.children1))
+            for (const child of filterTree.children1)
+                postProcessJsonLogicTree(child)
+        else if (isPlainObject(filterTree.children1))
+            for (const child of Object.values(filterTree.children1))
+                postProcessJsonLogicTree(child)
+    } else if (filterTree.type === "rule") {
+        const isLatest =
+            filterTree.properties.value.length === 1 &&
+            filterTree.properties.value[0] === "latest"
+        const isEarliest =
+            filterTree.properties.value.length === 1 &&
+            filterTree.properties.value[0] === "earliest"
+        const isEqual =
+            filterTree.properties.operator === "equal" ||
+            filterTree.properties.operator === "select_equals"
+        if (isLatest && isEqual) {
+            filterTree.properties.operator = "is_latest"
+        } else if (isEarliest && isEqual) {
+            filterTree.properties.operator = "is_earliest"
```

Comment:
I'd deconstruct to DRY this up a bit.

```
const { properties: { value }, operator }  = filterTree

if (value.length !== 1) return

const isLatest = value[0] === "latest"

const isEarliest = value[0] === "earliest"

const isEqual = operator === "equal" || operator === "select_equals"
```
Up to you! Still getting a sense for the preferred idioms on this project.

Reply:
Yeah true, that reads nicer. My brain always goes into legacy mode with these things in TS as in an ML language could write it like this:
```fsharp
match (filterTree.properties) with
| {value = ["latest"], operator = "equal"}
| {value = ["latest"], operator = "select_equal"} -> "is_latest"
| {value = ["earliest"], operator = "equal"}
| {value = ["earliest"], operator = "select_equal"} -> "is_earliest"
```